libs/pubsub/audit-logs/pubsub-bus.go

```diff
@@ -0,0 +1,132 @@
+package auditlog
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	buscore "github.com/twirapp/twir/libs/bus-core"
+	busauditlog "github.com/twirapp/twir/libs/bus-core/audit-logs"
+	"go.uber.org/fx"
+)
+
+type (
+	BusPubSub struct {
+		bus *buscore.Bus
+
+		subs       map[string]BusSubscription
+		subsLocker *sync.RWMutex
+	}
+
+	BusSubscription struct {
+		channel      chan AuditLog
+		dashboardIDs []string
+		done         chan struct{}
+	}
+)
+
+var (
+	_ PubSub       = (*BusPubSub)(nil)
+	_ Subscription = (*BusSubscription)(nil)
+)
+
+func NewBusPubSub(bus *buscore.Bus) BusPubSub {
+	return BusPubSub{
+		bus:  bus,
+		subs: make(map[string]BusSubscription),
```

Comment:
I think we need to init pointer to mutex here. So I think you didn't tested your code.